external-crates/move/crates/move-package-alt/src/package/package_impl.rs

```diff
@@ -272,41 +274,37 @@ impl<F: MoveFlavor> Package<F> {
     }
 
     /// Return the implicit deps depending on the implicit dep mode.
-    fn implicit_deps(
+    fn system_dependencies(
         env: &Environment,
-        implicit_dep_mode: ImplicitDepMode,
+        system_dependencies: Option<Vec<String>>,
     ) -> PackageResult<BTreeMap<PackageName, ReplacementDependency>> {
-        match implicit_dep_mode {
-            // For enabled state, we need to pick the deps based on whether there is
-            // a specfiied
-            ImplicitDepMode::Enabled(specified_deps) => {
-                let deps = F::implicit_deps(env.id().to_string());
-
-                if let Some(specified_deps) = specified_deps {
-                    // If a list of deps is specified, we need to make sure
-                    // that all of the deps are valid in the implicit deps list, or warn.
-                    for dep in &specified_deps {
-                        if !deps.contains_key(&Identifier::new(dep.as_str())?) {
-                            return Err(PackageError::Generic(format!(
-                                "The implicit dependency `{}` does not exist in the implicit deps list.",
-                                dep
-                            )));
-                        }
-                    }
-
-                    // If we have a "specified" list of deps, we need to filter the implicit deps to only support
-                    // the ones that are in the specified list.
-                    Ok(deps
-                        .into_iter()
-                        .filter(|(name, _)| specified_deps.contains(&name.to_string()))
-                        .collect())
+        if let Some(system_dependencies) = system_dependencies {
+            // The case were we do not want any system dependencies.
+            if system_dependencies.is_empty() {
```

Comment:
I don't think this should be a special case - it should just make the `for` loop run 0 iterations.